extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/BackChannelLogoutHandler.java

```diff
@@ -20,152 +18,127 @@
 import io.vertx.core.MultiMap;
 import io.vertx.core.http.HttpHeaders;
 import io.vertx.core.json.JsonObject;
-import io.vertx.ext.web.Router;
 import io.vertx.ext.web.RoutingContext;
 
-public class BackChannelLogoutHandler {
+public class BackChannelLogoutHandler implements Handler<RoutingContext> {
     private static final Logger LOG = Logger.getLogger(BackChannelLogoutHandler.class);
     private static final String SLASH = "/";
 
-    @Inject
-    DefaultTenantConfigResolver resolver;
-
-    private final OidcConfig oidcConfig;
-
-    public BackChannelLogoutHandler(OidcConfig oidcConfig) {
-        this.oidcConfig = oidcConfig;
-    }
-
-    public void setup(@Observes Router router) {
-        addRoute(router, oidcConfig.defaultTenant);
-
-        for (OidcTenantConfig oidcTenantConfig : oidcConfig.namedTenants.values()) {
-            addRoute(router, oidcTenantConfig);
-        }
-    }
+    private final DefaultTenantConfigResolver resolver;
+    private final OidcTenantConfig oidcTenantConfig;
 
-    private void addRoute(Router router, OidcTenantConfig oidcTenantConfig) {
-        if (oidcTenantConfig.isTenantEnabled() && oidcTenantConfig.logout.backchannel.path.isPresent()) {
-            router.route(oidcTenantConfig.logout.backchannel.path.get())
-                    .handler(new RouteHandler(oidcTenantConfig));
-        }
+    public BackChannelLogoutHandler(OidcTenantConfig oidcTenantConfig) {
+        this.oidcTenantConfig = oidcTenantConfig;
+        this.resolver = Arc.container().instance(DefaultTenantConfigResolver.class).get();
```

Comment:
I was not sure how to inject the `DefaultTenantConfigResolver` since I can't inject it into `OidcBuildStep#setup`,
so I used this because I saw it in:
https://github.com/quarkusio/quarkus/blob/aef1ab7d4c768f7db87fa998561e51b678820eb2/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcRecorder.java#L705

Reply:
@PhilKes It may require some more thinking, as far as I recall it is only there to get access to the map of the tenant specific back-channel cache. May be the cache should be in TenantConfigBean